api/v1alpha1/condition_types.go

```diff
@@ -0,0 +1,81 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConditionSeverity expresses the severity of a Condition Type failing.
+type ConditionSeverity string
+
+const (
+	// ConditionSeverityError specifies that a condition with `Status=False` is an error.
+	ConditionSeverityError ConditionSeverity = "Error"
+
+	// ConditionSeverityWarning specifies that a condition with `Status=False` is a warning.
+	ConditionSeverityWarning ConditionSeverity = "Warning"
+
+	// ConditionSeverityInfo specifies that a condition with `Status=False` is informative.
+	ConditionSeverityInfo ConditionSeverity = "Info"
+
+	// ConditionSeverityNone should apply only to conditions with `Status=True`.
+	ConditionSeverityNone ConditionSeverity = ""
+)
+
+// ConditionType is a valid value for Condition.Type.
+type ConditionType string
+
+// Condition defines an observation of a VM Operator API resource operational state.
+type Condition struct {
+	// Type of condition in CamelCase or in foo.example.com/CamelCase.
+	// Many .condition.type values are consistent across resources like Available, but because arbitrary conditions
+	// can be useful (see .node.status.conditions), the ability to deconflict is important.
+	// +required
+	Type ConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown.
+	// +required
+	Status corev1.ConditionStatus `json:"status"`
+
+	// Severity provides an explicit classification of Reason code, so the users or machines can immediately
+	// understand the current situation and act accordingly.
+	// The Severity field MUST be set only when Status=False.
+	// +optional
+	Severity ConditionSeverity `json:"severity,omitempty"`
+
+	// Last time the condition transitioned from one status to another.
+	// This should be when the underlying condition changed. If that is not known, then using the time when
+	// the API field changed is acceptable.
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// The reason for the condition's last transition in CamelCase.
+	// The specific API may choose whether or not this field is considered a guaranteed API.
+	// This field may not be empty.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// A human readable message indicating details about the transition.
+	// This field may be empty.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// Conditions provide observations of the operational state of a VM Operator API resource.
+type Conditions []Condition
```

Comment:
What's the benefit of having this type instead of using []Condition?

Reply:
Maybe little, but the imported cluster-api code uses this type and I don't think it is worth increasing our changes to remove it